pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -273,8 +273,10 @@ else let
        "checkInputs" "installCheckInputs"
        "__darwinAllowLocalNetworking"
        "__impureHostDeps" "__propagatedImpureHostDeps"
-       "sandboxProfile" "propagatedSandboxProfile"]
-       ++ lib.optional (__structuredAttrs || envIsExportable) "env"))
+       "sandboxProfile" "propagatedSandboxProfile"
+       "removeFromBuilderEnv"]
```

Comment:
With `__structuredAtts` the arguments passed to `mkDerivation` will not end up in the environment so this will become redundant.

Reply:
They would still become part of the JSON and part of the hash, causing rebuilds.
For instance, changes to `cargoUpdateHook` were previously masked by the FOD hash, but this reintroduces that value as an unnecessary input dependency that will cause a rebuild even if the FOD hash didn't have to change.

Similarly, you may want to have a "local" variable that only affects the passthru attributes, but isn't a passthru itself; perhaps some configuration for an automatically generated `updateScript`. That could only be expressed by filtering away at least one more attribute.

That said, perhaps the name could be improved by removing the `Env` part; what about `removeFromBuilder`?

It's not a pretty solution, but it's the best we can do when we don't have an alternative place for such attributes. We could have a single attribute like `locals` specifically for this purpose, but that doesn't help with attributes that already exist, and I don't like that it exposes an implementation detail to users. "Local" should be the default behavior of attributes at the root of the argument, but at that point it's not `mkDerivation` anymore.